hilti/toolchain/include/ast/operators/enum.h

```diff
@@ -24,17 +24,30 @@ STANDARD_OPERATOR_2x(
     type::Type_(type::UnsignedInteger(type::Wildcard())),
     "Casts an enum value into a unsigned integer. This will throw an exception if the enum value is ``Undef``.");
 
-BEGIN_CTOR(enum_, Ctor)
+BEGIN_CTOR(enum_, CtorSigned)
     auto ctorType() const { return type::Enum(type::Wildcard()); }
 
     auto signature() const {
-        return Signature{.args = {{.id = "value", .type = type::UnsignedInteger(type::Wildcard())}}, .doc = R"(
-Instantiates an enum instance initialized from an integer value. The value does
+        return Signature{.args = {{.id = "value", .type = type::SignedInteger(type::Wildcard())}}, .doc = R"(
+Instantiates an enum instance initialized from a signed integer value. The value does
 *not* need to correspond to any of the type's enumerator labels.
 )"};
     }
 END_CTOR
 
+BEGIN_CTOR(enum_, CtorUnsigned)
+    auto ctorType() const { return type::Enum(type::Wildcard()); }
+
+    auto signature() const {
+        return Signature{.args = {{.id = "value", .type = type::UnsignedInteger(type::Wildcard())}}, .doc = R"(
+Instantiates an enum instance initialized from an unsigned integer
+value. The value does *not* need to correspond to any of the type's
+enumerator labels. It must not be larger that the maximum that a
+*signed* 64-bit integer value can represent.
```

Comment:
This has a typo, needs to fix: than for that
Oh, and this clarifies that the limitation is not at the same size, so nevermind about my concern regarding 130 in a unit8. But this:
the maximum that a *signed* 64-bit integer value can represent
is a much more explanatory phrasing than the existing proposed phrasing in the .rst

Reply:
Ack, will fix.